fractal-terrain-generation/fractal_backend/fractal.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-console,no-bitwise */
+const noiseGen = require('./noiseGen');
+const simplify = require('./simplify');
+
+function makeHeaders(blockCount, maxHeight,
+  payloadBytes, genTime) {
+  const statusHeaders = {
+    'X-Gen-Data-Blockcount': blockCount,
+    'X-Gen-Data-Max-Height': maxHeight,
+    'X-Gen-Data-Payload-Bytes': payloadBytes,
+    'X-Gen-Data-Time-Running-MS': genTime,
+  };
+  const customHeaders = {
+    'Access-Control-Expose-Headers': Object.keys(statusHeaders).join(', '),
+    'Content-Type': 'application/octet-stream',
+    'Access-Control-Allow-Origin': '*',
+    'Access-Control-Allow-Headers': 'Origin X-Requested-With, Content-Type, Accept',
+    ... statusHeaders,
+  };
+  return customHeaders;
+}
+
+exports.handler = async (body, ctx) => {
+  const {
+    downscale,
+    heightFactor,
+    numTex,
+    size,
+    xPos,
+    yPos,
+    zPos,
+  } = ctx.request.query;
+
+  let respBody;
+  let headers;
+  let statusCode = 500;
+  const genStartTime = process.hrtime();
+
+  const { blockCount, data, maxHeight } = noiseGen(
+    parseInt(xPos, 10), parseInt(yPos, 10),
+    parseInt(zPos, 10), parseInt(size, 10),
+    numTex, downscale, parseInt(heightFactor, 10)
+  );
+  if (blockCount === 0) {
+    const failedGenTime = process.hrtime(genStartTime);
+    const genStr = (failedGenTime[0] * 1000) + (failedGenTime[1] / 1000000);
+    headers = makeHeaders(blockCount, maxHeight, 0, genStr);
+    respBody = new Buffer(2);
+    statusCode = 200;
```

Comment:
Code-style: my preference would be to have an early-return here (but would require moving into a separate function).